### PR TITLE
Special-case html validation for <li> without <ul>, <ol>, or <menu>

### PR DESCRIPTION
**Pull request by alexpelan**
Fixes: #515

### CODE REVIEW

File: src/validations/html/htmlInspector.js [hunk 1]
```diff
@@ -2,14 +2,36 @@ import last from 'lodash/last';
 import isNull from 'lodash/isNull';
 import Validator from '../Validator';
 
+const specialCases = {
+  li: {
+    reason: 'invalid-tag-parent',
+    parent: '<ul>, <ol>, or <menu>',
```
Comment (outoftime) [inline src/validations/html/htmlInspector.js]: Can we get the “or” of it all into the translation? Maybe `i18next` has a clever way of allowing you to construct a natural-language list out of an array?

File: src/validations/html/htmlInspector.js [hunk 2]
```diff
@@ -2,14 +2,36 @@ import last from 'lodash/last';
 import isNull from 'lodash/isNull';
 import Validator from '../Validator';
 
+const specialCases = {
+  li: {
```
Comment (outoftime) [inline src/validations/html/htmlInspector.js]: Any others come to mind? `<title>` inside `<head>` seems like a good one…

Comment (outoftime) [inline src/validations/html/htmlInspector.js]: (I’m down for that to come in later pull requests though)

File: src/validations/html/htmlInspector.js (hunk 1 above)
Comment (alexpelan) [inline src/validations/html/htmlInspector.js]: Couldn't figure out an easy way to do this, so I changed it to the more robotic sounding "The <li> tag needs to be inside one of the following tags: <ul>, <ol>, <menu>."

File: src/validations/html/htmlInspector.js (hunk 2 above)
Comment (alexpelan) [inline src/validations/html/htmlInspector.js]: Added title as a special case as well.

**Comment by alexpelan** [comment]
Rebased and pushed again

File: src/validations/html/htmlInspector.js (hunk 2 above)
Comment (jessewang-arvatosystems) [inline src/validations/html/htmlInspector.js]: Perhaps `<input>` inside `<form>` as well?

**Comment by outoftime** [comment]
@alexpelan what if we used something like [`arrayToSentence`](https://github.com/shinnn/array-to-sentence), and wrapped it in a function that passed in `options` based on a special key in the translation file?

**Comment by alexpelan** [comment]
I'm a little confused what you're looking for - so like, in Validator._convertErrorToAnnotation, pass a key in the payload to use the 'or' sentence logic?

**Comment by alexpelan** [comment]
@outoftime fixed the node version on npm-shrinkwrap. Should be ready for another look.

File: src/util/arrayToSentence.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import arrayToSentence from 'array-to-sentence';
+
+export function arrayToSentenceWithOr(array) {
+  return arrayToSentence(array, {lastSeparator: ' or '});
```
Comment (outoftime) [inline src/util/arrayToSentence.js]: Can we pull `' or '` out of the translation file, and call this something like `localizedArrayToSentence`?

**Pull request merged**